src/trace.rs

```diff
@@ -29,11 +29,21 @@ impl TracingRepresentation for crate::bson::oid::ObjectId {
     }
 }
 
-impl TracingRepresentation for crate::error::Error {
-    type Representation = String;
-
-    fn tracing_representation(&self) -> String {
-        self.to_string()
+impl crate::error::Error {
+    fn tracing_representation(&self, max_document_length: usize) -> String {
```

Comment:
Having to feature-conditionally thread this through everywhere is a bit unpleasant; what would you think about attaching `max_document_length` as a field of `Error` instead?